app/controllers/concerns/document_attachable.rb

```diff
@@ -25,6 +25,10 @@ def document_provided?
     tribunal_case&.documents(document_key)&.any? || document.present?
   end
 
+  def document_presence
```

Comment:
IMO for this simple validation, shuffling around the 1 line of code to another class (validator) will introduce a level of indirection that will make the code more difficult to follow.
I will refactor this and see how it looks, but might feel a bit over engineered tbh.